tests/integration/optlyplugins/test_composite_service.go

```diff
@@ -0,0 +1,80 @@
+package optlyplugins
+
+import (
+	"github.com/optimizely/go-sdk/pkg/decision"
+	"github.com/optimizely/go-sdk/pkg/notification"
+	"github.com/optimizely/go-sdk/tests/integration/models"
+)
+
+// TestCompositeService represents a CompositeService with custom implementations
+type TestCompositeService struct {
+	decision.CompositeService
+}
+
+// AddListener - Adds Notification Listeners
+func (c *TestCompositeService) AddListener(params models.APIOptions) (getListenersCalled func() []models.DecisionListener) {
+
+	var listenersCalled []models.DecisionListener
+	getListenersCalled = func() []models.DecisionListener {
+		return listenersCalled
+	}
+	if len(params.Listeners) < 1 {
+		return
+	}
+
+	for listenerType, count := range params.Listeners {
+		for i := 1; i <= count; i++ {
+			switch listenerType {
+			case "Decision":
+				callback := func(notification notification.DecisionNotification) {
```

Comment:
nit: To avoid so much nesting maybe define this function a a method on this struct and use it as such